### ISSUE TITLE
Bad error message when trying to construct an object incorrectly

### ISSUE THREAD

**Issue by dhasenan** [Araq/Nim#2584]
```
type Foo = object
  a: int
  b: float

discard Foo(1, 1.5)
```

This should produce an error message saying that `Foo` is a type and you should initialize it like `Foo(a: 1, b: 1.5)`.

The actual error message is: `badmsg.nim(5, 11) Error: undeclared identifier: 'Foo'`

**Comment by reactormonk** [comment]
That's because that is the proc calling syntax, it's looking for a `proc Foo`.

**Comment by dhasenan** [comment]
Right. But since it's saying there's no identifier named 'Foo' (which is simply false), I spent fifteen minutes wondering what was wrong with the declaration that meant the compiler couldn't find it.

A basic fix would simply replace the message with `procedure 'Foo' not found`. That's sufficient that an intelligent person moderately familiar with Nim would be able to think: "This isn't a procedure -- oh right, if it confused initialization with a procedure call, that means I wrote it wrong."

A good fix would search for any identifier named 'Foo' after failing to find any callable item with that name. If none was found, it would say `undeclared identifier 'Foo'`. If one was found, it wouldn't be callable (otherwise the thing would just compile), and it should give an error like `'Foo' is not callable`.

An excellent fix would go on to detect if `Foo` is a type, in which case it would prompt you to use the proper initialization syntax.

These would only slow the compiler down in the error case.
